tests/infrastructure/golden_images/update_boot_source/test_ssp_common_templates_boot_sources.py

```diff
@@ -59,35 +44,21 @@ def rhel9_data_source(golden_images_namespace):
 
 
 @pytest.fixture()
-def existing_data_source_volume(
-    golden_images_persistent_volume_claims_scope_function,
-    golden_images_volume_snapshot_scope_function,
-    matrix_data_source,
-):
-    source = matrix_data_source.source
-    if source.kind == PersistentVolumeClaim.kind:
-        cluster_volumes = golden_images_persistent_volume_claims_scope_function
-    else:
-        cluster_volumes = golden_images_volume_snapshot_scope_function
-    assert any([source.name in volume.name for volume in cluster_volumes]), (
-        f"DataSource source {source.kind} {source.name} is missing"
-    )
-    return matrix_data_source
-
-
-@pytest.fixture()
-def auto_update_boot_source_vm(
+def auto_update_boot_source_template_vm(
     unprivileged_client,
     namespace,
     existing_data_source_volume,
-    boot_source_os_from_data_source_dict,
+    boot_source_data_from_data_source_dict,
 ):
-    LOGGER.info(f"Create a VM using {existing_data_source_volume.name} dataSource")
+    data_source_name = existing_data_source_volume.name
+    if "template_os" not in boot_source_data_from_data_source_dict:
+        pytest.skip(f"{data_source_name} dont have a template")
```

Comment:
Why do we need to skip? Why do you expect boot_source_data_from_data_source_dict["template_os"] to not to be present?

Reply:
There are no templates available for CentOS 10 and RHEL 10, so we need to skip tests that rely on templates for these OS versions. I updated auto_update_data_source_matrix so that these two new data sources won't include the "template_os" field.

Reply:
Then you should not be auto_update_data_source_matrix  matrix. Please create and use a matrix that does not include `CentOS 10` or `RHEL 10`. No programmatic skips would be taken in main branch going forward.

Reply:
+1 for separate lists for template and non-template tests